### PR TITLE
Bikram/height adjustment

### PR DESCRIPTION
**Pull request by bikramkawan**
Please check on your side.
@thinkh @mijar @katush

### CODE REVIEW

**Comment by mstreit** [comment]
Works as expected, thanks @bikramkawan. 
@thinkh, can you look over the code and merge it please?

File: src/column/ColumnManager.ts [hunk 1]
```diff
@@ -227,12 +236,43 @@ export default class ColumnManager extends EventHandler {
...
+  private async calColHeight(height) {
+
+
+    const vectorColumns = this.columns.filter((d) => d.data.desc.type === AColumn.DATATYPE.vector);
+    const dataPoints = [];
+    for (const r of this.rangeList) {
+      const view = await vectorColumns[0].data.idView(r);
+      dataPoints.push(await (<IAnyVector>view).length);
+    }
+    const cols = [];
```
Comment (thinkh) [inline src/column/ColumnManager.ts]:
```typescript
const cols = vectorColumns.map((col) => {
 return dataPoints.map((d) => {
   return {minHeight: col.minHeight * d, maxHeight: col.maxHeight * d};
 }
});

const colsFlatten =cols.reduce(
  (acc, val) => acc.concat(
    Array.isArray(val) ? flatten(val) : val
  ),
  []
); 
```

File: src/column/ColumnManager.ts [hunk 2]
```diff
@@ -227,12 +236,43 @@ export default class ColumnManager extends EventHandler {
...
+    for (const col of vectorColumns) {
+      for (const d of dataPoints) {
+        cols.push({minHeight: col.minHeight * d, maxHeight: col.maxHeight * d});
+      }
+    }
+    const minHeight = d3.max(cols, (d, i) => d.minHeight);
```
Comment (thinkh) [inline src/column/ColumnManager.ts]:
```typescript
const minHeight = Math.max(...cols.map((d) => d.minHeight)))
```

File: src/column/ColumnManager.ts [hunk 3]
```diff
@@ -227,12 +236,43 @@ export default class ColumnManager extends EventHandler {
...
+    const maxHeight = d3.min(cols, (d, i) => d.maxHeight);
+    const checkStringCol = vectorColumns.filter((d) => (<any>d).data.desc.value.type === VALUE_TYPE_STRING);
+
+    if (checkStringCol.length > 0 && minHeight > height) {
```
Comment (thinkh) [inline src/column/ColumnManager.ts]:
```typescript
if(checkStringCol.length > 0) {
  minHeight = Math.max(maxHeight, minHeight);
}
return Math.max(minHeight, Math.min(maxHeight, height));
```

File: src/column/ColumnManager.ts [hunk 4]
```diff
@@ -197,9 +202,13 @@ export default class ColumnManager extends EventHandler {
 
   async update(idRange: Range) {
     this.rangeNow = idRange;
+    this.rangeList = [];
```
Comment (thinkh) [inline src/column/ColumnManager.ts]: Overriding a list and pushing just one element is not useful.

**Comment by thinkh** [comment]
Thanks @bikramkawan

**Pull request merged**